src/ScheduledTask/Webhook/AuthorisationWebhookHandler.php

```diff
@@ -97,27 +106,39 @@ private function handleSuccessfulNotification(
         OrderTransactionEntity $orderTransaction,
         NotificationEntity $notification,
         Context $context
-    ) {
+    ): void {
         $paymentMethodHandler = $orderTransaction->getPaymentMethod()->getHandlerIdentifier();
+        $isManualCapture = $this->captureService->requiresManualCapture($paymentMethodHandler);
+        $currencyUtil = new Currency();
+        $totalPrice = $orderTransaction->getAmount()->getTotalPrice();
+        $isoCode = $orderTransaction->getOrder()->getCurrency()->getIsoCode();
+        $transactionAmount = $currencyUtil->sanitize($totalPrice, $isoCode);
 
-        if ($this->captureService->requiresManualCapture($paymentMethodHandler)) {
-            $this->logger->info(
-                'Manual capture required. Setting payment to `authorised` state.',
-                ['notification' => $notification->getVars()]
-            );
-            $this->orderTransactionStateHandler->authorize($orderTransaction->getId(), $context);
+        $this->adyenPaymentService->insertAdyenPayment($notification, $orderTransaction, $isManualCapture);
 
-            $this->logger->info(
-                'Attempting capture for open invoice payment.',
-                ['notification' => $notification->getVars()]
-            );
-            $this->captureService->doOpenInvoiceCapture(
-                $notification->getMerchantReference(),
-                $notification->getAmountValue(),
-                $context
-            );
-        } else {
-            $this->orderTransactionStateHandler->paid($orderTransaction->getId(), $context);
+        if ($transactionAmount == $notification->getAmountValue()) {
```

Comment:
@peterojo could you double-check this change of moving the amount comparison before going into processing of the AUTHORISATION notification?